### PR TITLE
Entity shower story and documentation

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | -- |
| Dependencies | https://github.com/ripe-tech/ripe-components-vue/pull/383 |
| Decisions | • Created `entity-shower` storybook story <br>• Documented `entity-shower` props |
| Animated GIF | Captura de ecrã 2020-10-08 às 11 33 50 |

### CODE REVIEW

File: vue/components/ui/organisms/entity-shower/entity-shower.vue [hunk 1]
```diff
@@ -28,50 +28,91 @@ export const EntityShower = {
     name: "entity-shower",
     mixins: [partMixin],
     props: {
+        /**
+         * The entity name
```
Comment (gcandal) [inline vue/components/ui/organisms/entity-shower/entity-shower.vue:32]: JSDocs - unlike comments - end with a period (`.`).

Comment (gcandal) [reply vue/components/ui/organisms/entity-shower/entity-shower.vue:32]: Try by a little more descriptive. What is the entity name? Can it be `João` or should it be `Person`? Use examples if  necessary.

File: vue/components/ui/organisms/entity-shower/entity-shower.vue [hunk 2]
```diff
@@ -28,50 +28,91 @@ export const EntityShower = {
...
     props: {
+        /**
+         * The entity name
+         */
         name: {
             type: String,
             required: true
         },
+        /**
+         * Entity fields that are going to be shown
```
Comment (gcandal) [inline vue/components/ui/organisms/entity-shower/entity-shower.vue:39]: Be more descriptive, from the docs I can't understand what must be passed here without checking the implementation.

File: vue/components/ui/organisms/entity-shower/entity-shower.vue [hunk 3]
```diff
@@ -28,50 +28,91 @@ export const EntityShower = {
...
+         * Entity's delete method. If set, a delete option is shown in
+         * the options dropdown
+         */
         deleteEntity: {
             type: Function,
             default: null
         },
+        /**
+         * Message shown when an error occurs
+         * the options dropdown
```
Comment (gcandal) [inline vue/components/ui/organisms/entity-shower/entity-shower.vue:107]: ?

File: vue/components/ui/organisms/entity-shower/entity-shower.vue [hunk 4]
```diff
@@ -28,50 +28,91 @@ export const EntityShower = {
...
+         */
         getErrorMessage: {
             type: Function,
             default: error => error.message || "Something went wrong"
         },
+        /**
+         * Props that can be used to customize the error notification
```
Comment (gcandal) [inline vue/components/ui/organisms/entity-shower/entity-shower.vue:114]: If you call it `errorMessage` then it must be "...the error message"

File: vue/components/ui/organisms/entity-shower/entity-shower.vue (hunk 2 above)
Comment (gcandal) [reply vue/components/ui/organisms/entity-shower/entity-shower.vue:39]: In general this applies to most docs. Always think to yourself that if a doc is enough to use the prop without checking the rest of the code.

**Review by gcandal — CHANGES_REQUESTED** [review]

File: vue/components/ui/organisms/entity-shower/entity-shower.vue (hunk 3 above)
Comment (NFSS10) [reply vue/components/ui/organisms/entity-shower/entity-shower.vue:107]: Probably merge problem :S

File: vue/components/ui/organisms/entity-shower/entity-shower.vue (hunk 2 above)
Comment (NFSS10) [reply vue/components/ui/organisms/entity-shower/entity-shower.vue:39]: Should I also add the prop type in the documentation?
I tried to follow the comments in `pickers` and `configurator`

Comment (gcandal) [reply vue/components/ui/organisms/entity-shower/entity-shower.vue:39]: No need, but you can add an example (e.g.: [{ name: ... , ... }])

**Comment by NFSS10** [comment]
@gcandal Please check if it's good enough or if it still needs more explanation and/or examples

**Pull request merged**